content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,327 @@
+---
+title: 'Referential actions'
+metaTitle: 'Referential actions'
+metaDescription: 'Referential actions let you define the update and delete behavior of related models on the database level'
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+As of **2.26.0** Prisma now supports referential actions as part of the Prisma Schema Language.
+
+This is a [**preview feature**](../../../about/releases#preview) and can be accessed by:
+
+1. [Upgrading to version v2.26.0 of Prisma](../../../guides/upgrade-guides/upgrading-to-latest).
+2. Adding `referentialActions` to the `previewFeatures` field of the [`generator`](../../../reference/api-reference/prisma-schema-reference#generator) block in the `prisma.schema` file.
+
+```prisma file=schema.prisma
+generator client {
+    provider = "prisma-client-js"
+    previewFeatures: ['referentialActions']
+}
+```
+
+You now have 3 options:
+
+1. Make your schema reflect the current state of the database by running an [Introspection](../introspection). You can now see all relations in your data model using a different referential action compared to the new defaults.
+2. Define custom referential actions on your data model relation fields. You then update the database schema by running a [Migration](../prisma-migrate) or the [`db push`](../prisma-migrate/db-push) command.
+3. Use the new default referential actions that exist when no custom actions are added manually by running a [Migration](../prisma-migrate), or [`db push`](../prisma-migrate/db-push).
+
+To begin using referential actions you add them to the `@relation` attribute in your schema.
+
+```prisma
+@relation(fields: [authorId], references: [id], onDelete: Cascade)
+```
+
+</TopBlock>
+
+## What are referential actions?
+
+Referential actions are constraints that can be used to define how a referenced record is handled by the database when either an [`update`](../prisma-client/crud#update) or [`delete`](../prisma-client/crud#delete) query is run.
+
+These include:
+
+- [`Cascade`](#cascade)
+- [`Restrict`](#restrict)
+- [`NoAction`](#noaction)
+- [`SetNull`](#setnull)
+- [`SetDefault`](#setdefault)
+
+To understand how referential actions can impact your data it is important to understand how foreign key constraints work.
```

Comment:
Is it important to understand how foreign keys work to understand Prisma relations? If not, I am questioning if this here is really different. We should be able to explain this concept from the perspective of Prisma schema features (relations), and not the underlying SQL features (foreign keys).

Reply:
But isnt knowing about the how the foreign keys are related to how relations work an important aspect of understanding why the changes to the default ref actions work? I mean, its good to get a brief overview, I don't go into detail

Reply:
I don't think it is necessary though. Prisma had fields, some of them can be relations, the relations can define referential actions that do certain things.

How that works under the hood can also be documented, but is not necessary to understand the fundemantal concept and how Prisma works. We usually try to abstract that away from the user, and only mention it to the people that really care or need to know it for some reason. Here this sounds different.